src/renderer/coremods/contextMenu/index.tsx

```diff
@@ -0,0 +1,121 @@
+import type {
+  ContextItem,
+  ContextMenuTypes,
+  GetContextItem,
+  RawContextItem,
+} from "../../../types/coremods/contextMenu";
+import { Logger } from "../../modules/logger";
+import { getByProps } from "../../modules/webpack";
+import { React } from "@common";
+
+const logger = Logger.api("ContextMenu");
+
+export const menuItems = {} as Record<ContextMenuTypes, GetContextItem[]>;
+
+/**
+ * Converts data into a react element. Any elements or falsy value will be returned as is
+ * @param raw The data to convert
+ * @returns The converted item
+ */
+function makeItem(raw: RawContextItem | ContextItem | undefined): ContextItem | undefined {
+  // Occasionally React won't be loaded when this function is ran, so we don't return anything
+  if (!React) return undefined;
+
+  if (!raw) {
+    // If something falsy is passed, let it through
+    // Discord just skips over them too
+    return raw;
+  }
+  if (React.isValidElement(raw)) {
+    // We can't construct something that's already made
+    return raw as unknown as ContextItem;
+  }
+
+  raw = raw as RawContextItem;
```

Comment:
ESLint says this is unnecessary, please double check if that's the case and either remove this or ignore the warning.